rest_framework/validators.py

```diff
@@ -212,6 +215,17 @@ def enforce_required_fields(self, attrs):
         if missing_items:
             raise ValidationError(missing_items, code='required')
 
+    def _get_field_values(self, attrs):
```

Comment:
Perhaps just `get_value_and_date`?

Wondering if something like this might make the intent more clear?...

```python
instance = getattr(self, "instance", None)
if instance is not None:
    default_value = getattr(self.instance, self.field)
    default_date = getattr(self.instance, self.date_field)
    value = attrs.get(self.field, default_value)
    date = attrs.get(self.date_field, default_date)
else:
    value = attrs[self.field]
    date = attrs[self.date_field]

return (value, date)
```

Reply:
Thanks for taking a look at this! I've made the requested changes.

Reply:
Yup & yup!